src/main/java/net/floodlightcontroller/statistics/web/BandwidthResource.java

```diff
@@ -66,7 +67,9 @@ public Object retrieve() {
                 spbs = new HashSet<SwitchPortBandwidth>(Collections.singleton(statisticsService.getBandwidthConsumption(dpid, port)));
             } else {
                 spbs = new HashSet<SwitchPortBandwidth>();
-                for (OFPortDesc pd : switchService.getSwitch(dpid).getPorts()) { /* do specific DPID; do all ports */
+                //fix concurrency scenario
+                IOFSwitch sw = switchService.getSwitch(dpid);
+                for (OFPortDesc pd : sw.getPorts()) { /* do specific DPID; do all ports */
```

Comment:
@rizard yes, you are right. I am thinking what kind of error message should Statistics module return in case of the retrieved switch is offline. Can we return a null to retrieve() functions? Do you have any suggestion?